### PR TITLE
unzip with gz too

### LINKED ISSUE #4876: [suggestion] Support .gz files in tools.get()
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

I suggest that `tools.get()` supports .gz files in the same way as .tar.gz.

I want to download a gzipped patch and apply it.
https://github.com/nkshigeru/conan-libpng/blob/cdebdf4ae4a7b33871b6080ac13927b1036bdd41/conanfile.py#L45-L53

`tools.get()` is a very useful function but it does not support .gz file.
So I need to use low level functions.
```
tools.download("http://example.com/example.patch.gz", "example.patch.gz")
tools.check_sha1("example.patch.gz", hash)
with gzip.open("example.patch.gz", "rb") as f:
    patch = f.read().decode()
    tools.patch(patch_string=patch)
```
It would be great if I could write as below.
```
tools.get("http://example.com/example.patch.gz", sha1=hash)
# we get example.patch
tools.patch(patch_file="example.path")
```

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: ``tools.get()`` and ``tools.unzip()`` now handle also ``.gz`` compressed files
Docs: https://github.com/conan-io/docs/pull/1230

Close #4876

@tags: slow

### CODE REVIEW

**Comment by memsharded** [comment]
> Docs omit? don't be lazy please: https://docs.conan.io/en/latest/reference/tools.html#tools-unzip

Yes, sorry, I did a (very) quick look at the docs, and thought the extensions were not detailed. Docs added.

File: conans/client/tools/files.py [hunk 1]
```diff
@@ -70,6 +70,13 @@ def unzip(filename, destination=".", keep_permissions=False, pattern=None, outpu
             filename.endswith(".tbz2") or filename.endswith(".tar.bz2") or
             filename.endswith(".tar")):
         return untargz(filename, destination, pattern)
+    if filename.endswith(".gz"):
+        import gzip
+        with gzip.open(filename, 'rb') as f:
+            file_content = f.read()
+        target_name = filename[:-3] if destination == "." else destination
```
Comment (jgsogo) [inline conans/client/tools/files.py]: Here the destination is expected to be a file (not a directory like in the other cases), the docstring for this function doesn't match. If given a directory this function will fail in the `load` with a `IsADirectoryError`. I'm not sure if we want to handle it better.

Comment (memsharded) [reply conans/client/tools/files.py]: Updated docstring.

I have tried passing a folder as argument, and it failed (py2, Win), with ``IOError: [Errno 13] Permission denied: 'mytemp'``. I think in either case the error should be clear enough, catching and re-raising a ConanException is a bit overkill for this case.

Comment (jgsogo) [reply conans/client/tools/files.py]: No, don't need to reraise, it was just about the docstring (it is a documented tool). Don't forget to update it: https://docs.conan.io/en/latest/reference/tools.html#tools-unzip 😉

**Pull request merged**